comps/llms/text-generation/vllm/langchain/llm.py

```diff
@@ -127,11 +127,13 @@ async def stream_generator():
                 # use rag default template
                 prompt = ChatTemplate.generate_rag_prompt(input.query, input.documents)
 
+        new_input = LLMParamsDoc(query=prompt)
+
         if input.streaming:
 
             async def stream_generator():
                 chat_response = ""
-                async for text in llm.astream(input.query, **parameters):
+                async for text in llm.astream(new_input.query, **parameters):
```

Comment:
instead of reassigned prompt into new_input.query we can directly pass prompt variable into right?